vscode-dotty/src/extension.ts

```diff
@@ -19,7 +19,16 @@ let extensionContext: ExtensionContext
 let outputChannel: vscode.OutputChannel
 
 /** The sbt process that may have been started by this extension */
-let sbtProcess: ChildProcess
+let sbtProcess: ChildProcess | undefined
+
+/** The status bar where the show the status of sbt server */
+let sbtStatusBar: vscode.StatusBarItem
+
+/** Interval in ms to check that sbt is alive */
+const sbtCheckIntervalMs = 10 * 1000
+
+/** A command that we use to check that sbt is still alive. */
+export const nopCommand = "nop"
```

Comment:
I'm not sure this is the way to do things:
- If the connection with sbt is closed, we don't need this since we can have a callback on connection.onClose
- If sbt blocks while running something, then we will never get a response since there's no timeout for a response
- We can't add a timeout for the response, because an sbt command can take an arbitrarily long time to finish (e.g. running tests) and sbt cannot execute more than one command at once.

I think a real solution will require changes to sbt, but I don't know what exactly.